### PR TITLE
feat!: adopt 6.x driver MONGOSH-1541

### PR DESCRIPTION
**Pull request by addaleax**

### CODE REVIEW

**Comment by addaleax** [comment]
First Evergreen run (expecting a lot of red): https://spruce.mongodb.com/version/64dbaa6d32f417ddeb448895/tasks

**Comment by addaleax** [comment]
Evergreen: https://spruce.mongodb.com/version/64ec6fc6d6d80a2acd50bc88/tasks

**Comment by addaleax** [comment]
Evergreen: https://spruce.mongodb.com/version/64ec78745623436448263e6e/tasks

File: .evergreen/install-npm-deps.sh [hunk 1]
```diff
@@ -1,7 +1,7 @@
 set -e
 set -x
 
-npm ci --verbose
+npm ci --verbose --legacy-peer-deps
```
Comment (addaleax) [inline .evergreen/install-npm-deps.sh:4]: Leaving this comment as a reminder to myself to revert aee60aa5c532e0f3f3ba154aec76639879c83243 later

**Comment by addaleax** [comment]
Evergreen: https://spruce.mongodb.com/version/64ec91ba1e2d17a71146e70f/tasks

**Comment by addaleax** [comment]
Evergreen: https://spruce.mongodb.com/version/64eca9800ae6062fc62e341f/tasks

**Comment by addaleax** [comment]
Evergreen: https://spruce.mongodb.com/version/64ed1b8e32f417f18ce916f2/tasks

**Comment by addaleax** [comment]
Evergreen: https://spruce.mongodb.com/version/64edce689ccd4ea20b3ab4f2/tasks

**Comment by addaleax** [comment]
Evergreen: https://spruce.mongodb.com/version/64ede3ad850e61f91b02573d/tasks

**Review by lerouxb — APPROVED** [review]

**Pull request merged**